src/api/util.js

```diff
@@ -30,11 +30,23 @@ import {
   CardanoTxSigningMode,
 } from '../../temporary_modules/trezor-connect/';
 
-export const blockfrostRequest = async (endpoint, headers, body) => {
+export async function delay(delayInMs) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, delayInMs);
+  });
+}
+
+export async function blockfrostRequest(endpoint, headers, body) {
   const network = await getNetwork();
+  let result;
 
-  return await new Promise(async (res, rej) => {
-    const result = await fetch(provider.api.base(network.node) + endpoint, {
+  while (!result || result.status_code === 500) {
+    if (result) {
+      await delay(100);
+    }
```

Comment:
We’ll loop while the status_code is 500. The first request won’t wait 100 milliseconds, but all of the other ones will.